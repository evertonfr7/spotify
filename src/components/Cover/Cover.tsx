import Typography from '../ui/Typography'
import { CoverProps } from './Cover.types'

export function Cover({ id, image, title, subtitle }: CoverProps) {
  return (
    <div className="flex items-center gap-2 md:gap-4 text-white" key={id}>
      <img
        src={image}
        alt={title}
        aria-label={title}
        width={72}
        height={72}
        className="w-[48px] h-[48px] md:w-[72px] md:h-[72px]"
      />
      <div>
        <Typography className="font-normal">{title}</Typography>
        <Typography className="font-normal text-white/80 text-[10px]/[18px] md:text-[12px]/[20px]">
          {subtitle}
        </Typography>
      </div>
    </div>
  )
}
