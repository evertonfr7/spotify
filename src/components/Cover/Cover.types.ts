export interface CoverProps {
  id: string
  image: string
  title: string
  subtitle: string
}
