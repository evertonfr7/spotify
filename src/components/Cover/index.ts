export { Cover as default } from './Cover'
