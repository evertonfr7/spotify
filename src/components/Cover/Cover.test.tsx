import { render, screen } from '@testing-library/react'
import { Cover } from './Cover'

describe('Cover Component', () => {
  const mockProps = {
    id: '1',
    image: 'test-image.jpg',
    title: 'Test Title',
    subtitle: 'Test Subtitle',
  }

  test('renders the cover with correct snapshot', () => {
    const tree = render(<Cover {...mockProps} />)

    expect(tree).toMatchSnapshot()
  })

  test('renders the component with the provided props', () => {
    render(<Cover {...mockProps} />)

    const imageElement = screen.getByRole('img', { name: mockProps.title })
    expect(imageElement).toBeInTheDocument()
    expect(imageElement).toHaveAttribute('src', mockProps.image)
    expect(imageElement).toHaveAttribute('alt', mockProps.title)

    const titleElement = screen.getByText(mockProps.title)
    expect(titleElement).toBeInTheDocument()
    expect(titleElement).toHaveClass('font-normal')

    const subtitleElement = screen.getByText(mockProps.subtitle)
    expect(subtitleElement).toBeInTheDocument()
    expect(subtitleElement).toHaveClass('text-white/80')
  })

  test('applies responsive classes correctly', () => {
    render(<Cover {...mockProps} />)

    const imageElement = screen.getByRole('img', { name: mockProps.title })
    expect(imageElement).toHaveClass(
      'w-[48px] h-[48px] md:w-[72px] md:h-[72px]',
    )

    const subtitleElement = screen.getByText(mockProps.subtitle)
    expect(subtitleElement).toHaveClass(
      'text-[10px]/[18px] md:text-[12px]/[20px]',
    )
  })
})
