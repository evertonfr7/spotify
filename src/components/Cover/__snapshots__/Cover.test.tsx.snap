// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cover Component renders the cover with correct snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex items-center gap-2 md:gap-4 text-white"
      >
        <img
          alt="Test Title"
          aria-label="Test Title"
          class="w-[48px] h-[48px] md:w-[72px] md:h-[72px]"
          height="72"
          src="test-image.jpg"
          width="72"
        />
        <div>
          <p
            class="font-default text-[12px]/[18px] text-white md:text-[14px]/[20px] font-normal"
          >
            Test Title
          </p>
          <p
            class="font-default font-normal text-white/80 text-[10px]/[18px] md:text-[12px]/[20px]"
          >
            Test Subtitle
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-center gap-2 md:gap-4 text-white"
    >
      <img
        alt="Test Title"
        aria-label="Test Title"
        class="w-[48px] h-[48px] md:w-[72px] md:h-[72px]"
        height="72"
        src="test-image.jpg"
        width="72"
      />
      <div>
        <p
          class="font-default text-[12px]/[18px] text-white md:text-[14px]/[20px] font-normal"
        >
          Test Title
        </p>
        <p
          class="font-default font-normal text-white/80 text-[10px]/[18px] md:text-[12px]/[20px]"
        >
          Test Subtitle
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
